### ISSUE TITLE
Sometimes Hyprland advertises mirrored external monitor. This makes grim sad.

### ISSUE THREAD

**Issue by 875d** [hyprwm/Hyprland#1471]
I have an external monitor and this in my config file:

```
# MONITORS
# See https://wiki.hyprland.org/Configuring/Monitors/
monitor=eDP-1,preferred,auto,1
monitor=HDMI-A-1,preferred,auto,1,mirror,eDP-1
```

Normally Hyprland only advertise the laptop monitor (the mirrored one):
```
interface: 'zxdg_output_manager_v1',                     version:  3, name: 21
	xdg_output_v1
		output: 45
		name: 'eDP-1'
		description: 'AU Optronics 0xAF90 (eDP-1)'
		logical_x: 0, logical_y: 0
		logical_width: 1920, logical_height: 1080
```

But sometimes it decides it is proper to advertise both of them:
```
interface: 'zxdg_output_manager_v1',                     version:  3, name: 21
	xdg_output_v1
		output: 47
		name: 'HDMI-A-1'
		description: 'Samsung Electric Company C24F390 H4LT305839 (HDMI-A-1)'
		logical_x: -1337420, logical_y: -1337420
		logical_width: 1920, logical_height: 1080
	xdg_output_v1
		output: 45
		name: 'eDP-1'
		description: 'AU Optronics 0xAF90 (eDP-1)'
		logical_x: 0, logical_y: 0
		logical_width: 1920, logical_height: 1080
```

Since the external monitor has very low logical position, and since grim tries to capture both of them, grim computes an image size that far exceeds available memory and crashes.

When, in the future, it will not crash, it will error out smoothly without supporting this use case.

See [grim issue 85](https://todo.sr.ht/~emersion/grim/85).

**Comment by 875d** [comment]
Do you need some other logs to fix this?

I often cannot take screenshots when I'm connected to an external monitor

**Comment by vaxerski** [comment]
[patch.txt](https://github.com/hyprwm/Hyprland/files/10675004/patch.txt)

try this patch

**Comment by 875d** [comment]
Do you know how can I consistently reproduce this, in order to better test the patch?

**Comment by vaxerski** [comment]
nope

**Comment by vaxerski** [comment]
any news?

**Comment by 875d** [comment]
Sorry I don't use Hyprland right now

**Comment by vaxerski** [comment]
possibly fixed with 1ead6c46f4742c73311071dbae31caa66e40a323
